pkg/image/registry/imagestream/etcd/etcd.go

```diff
@@ -139,6 +143,72 @@ func (r *InternalREST) Update(ctx context.Context, name string, objInfo rest.Upd
 	return r.store.Update(ctx, name, objInfo, createValidation, updateValidation)
 }
 
+// LayersREST implements the REST endpoint for changing both the spec and status of an image stream.
+type LayersREST struct {
+	store *registry.Store
+	index ImageLayerIndex
+}
+
+var _ rest.Getter = &LayersREST{}
+
+func (r *LayersREST) New() runtime.Object {
+	return &imageapi.ImageStreamLayers{}
+}
+
+// Get returns the layers for an image stream.
+func (r *LayersREST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	if !r.index.HasSynced() {
+		return nil, errors.NewServerTimeout(r.store.DefaultQualifiedResource, "get", 2)
+	}
+	obj, err := r.store.Get(ctx, name, options)
+	if err != nil {
+		return nil, err
+	}
+	is := obj.(*imageapi.ImageStream)
+	isl := &imageapi.ImageStreamLayers{
+		ObjectMeta: is.ObjectMeta,
```

Comment:
The contract on this resource version is going to be broken.  GET of the same resource version can return two different objects.  Can we clear it?  Set it to something non-numeric?

Reply:
> The contract on this resource version is going to be broken. GET of the same resource version can return two different objects. Can we clear it? Set it to something non-numeric?

or a tuple of this and the image cache maybe?

Reply:
Because images are static, an image stream at a specific point in time (resource version) should always return the same layers, modulo someone deleting the image out of etcd manually.